test/Datadog.Trace.ClrProfiler.IntegrationTests/SerilogTests.cs

```diff
@@ -0,0 +1,80 @@
+// </copyright>
+
+using System;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class SerilogTests : LogsInjectionTestBase
+    {
+        private readonly LogFileTest[] _log200FileTests =
+            {
+                new LogFileTest()
+                {
+                    FileName = "log-textFile.log",
+                    RegexFormat = @"{0}: ""{1}""",
+                    PropertiesAreAlwaysPresent = true,
+                    PropertiesUseSerilogNaming = true
+                },
+                new LogFileTest()
+                {
+                    FileName = "log-jsonFile.log",
+                    RegexFormat = @"""{0}"":""{1}""",
+                    PropertiesAreAlwaysPresent = false,
+                    PropertiesUseSerilogNaming = true
+                }
+            };
+
+        private readonly LogFileTest[] _logPre200FileTests =
+            {
+                new LogFileTest()
+                {
+                    FileName = "log-textFile.log",
+                    RegexFormat = @"{0}: ""{1}""",
+                    PropertiesAreAlwaysPresent = true,
+                    PropertiesUseSerilogNaming = true
+                }
+            };
+
+        public SerilogTests(ITestOutputHelper output)
+            : base(output, "LogsInjection.Serilog")
+        {
+            SetServiceVersion("1.0.0");
+        }
+
+        [Theory]
+        [MemberData(nameof(PackageVersions.Serilog), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [Trait("Category", "LinuxUnsupported")]
+        public void InjectsLogs(string packageVersion)
+        {
+            SetEnvironmentVariable("DD_LOGS_INJECTION", "true");
+
+            int agentPort = TcpPortProvider.GetOpenPort();
+            using (var agent = new MockTracerAgent(agentPort))
+            using (var processResult = RunSampleAndWaitForExit(agent.Port, packageVersion: packageVersion))
+            {
+                Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+
+                var spans = agent.WaitForSpans(1, 2500);
+                Assert.True(spans.Count >= 1, $"Expecting at least 1 span, only received {spans.Count}");
+
+                if (string.IsNullOrWhiteSpace(packageVersion) || new Version(packageVersion) >= new Version("2.0.0"))
+                {
+                    ValidateLogCorrelation(spans, _log200FileTests);
+                }
+                else
+                {
+                    // We do not expect logs injection for Serilog versions < 2.0.0 so filter out all logs
+                    ValidateLogCorrelation(spans, _logPre200FileTests, additionalInjectedLogFilter: (_) => false);
```

Comment:
I wonder if we should explicitly assert that we _don't_ have logs injection in this case? Otherwise it's not confirming that the noop logger is used?

Reply:
Good idea. Done in ebbe57057684f7e18e04146cba343448d8d3c64c . I figured we should probably also test that we don't have correlated logs when DD_LOGS_INJECTION=false, so I tested that as well